Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+                return true
+            }
+        }
+        return false
+    }
+
+    @discardableResult @objc func start() -> Bool {
+        // Gather potential renderer discoverers
+        guard let tmpDiscoverers: [VLCRendererDiscovererDescription] = VLCRendererDiscoverer.list() else {
+            return false
+        }
+        for discoverer in tmpDiscoverers {
+            if !isDuplicateDiscoverer(with: discoverer) {
+                if let rendererDiscoverer = VLCRendererDiscoverer(name: discoverer.name) {
+                    if rendererDiscoverer.start() {
+                        rendererDiscoverer.delegate = self
+                        discoverers.append(rendererDiscoverer)
+                    } else {
+                        print("Unable to start renderer discoverer with name: \(rendererDiscoverer.name)")
+                    }
+                } else {
+                    print("Unable to instanciate renderer discoverer with name: \(discoverer.name)")
+                }
+            }
+        }
+
+        return true
+    }
+
+    @objc func stop() {
+        for discoverer in discoverers {
+            discoverer.stop()
+        }
+        discoverers.removeAll()
+    }
+
+    // MARK: VLCActionSheet
+    @objc fileprivate func displayActionSheet() {
+        if let presentingViewController = presentingViewController {
+            // If only one renderer, choose it automatically
+            if getAllRenderers().count == 1 {
+                let indexPath = IndexPath(row: 0, section: 0)
+                if let rendererItem = getAllRenderers().first {
+                    actionSheet.collectionView.selectItem(at: indexPath, animated: false, scrollPosition: .centeredVertically)
+                    actionSheet(collectionView: actionSheet.collectionView, didSelectItem: rendererItem, At: indexPath)
+                    setRendererItem(rendererItem: rendererItem)
+
+                    if let movieViewController = presentingViewController as? VLCMovieViewController {
+                        movieViewController.setupCastWithCurrentRenderer()
+                    }
+                }
+            } else {
+                presentingViewController.present(actionSheet, animated: false, completion: nil)
+            }
+        } else {
+            print("VLCRendererDiscovererManager: Cannot display actionSheet, no viewController setted")
+        }
+    }
+
+    fileprivate func setRendererItem(rendererItem: VLCRendererItem) {
+        let vpcRenderer = VLCPlaybackController.sharedInstance().renderer
+
+        if vpcRenderer != rendererItem {
+            VLCPlaybackController.sharedInstance().renderer = rendererItem
+            for button in rendererButtons {
+                button.isSelected = true
+            }
+        } else {
+            // Same renderer selected, removing selection
+            VLCPlaybackController.sharedInstance().renderer = nil
+            for button in rendererButtons {
```

Comment:
you shouldn't need to iterate over all of them but be able to disable the one that was selected